lib/rubocop/cop/rspec/no_expectation_example.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks if an example includes any expectation.
+      #
+      # All RSpec's example and expectation methods are covered by default.
+      # If you are using your own custom methods,
+      # add the following configuration:
+      #
+      #   RSpec/NoExpectationExample:
+      #     ExampleMethodnames:
+      #       - custom_it
+      #     ExpectationMethodNames:
+      #       - custom_expect
+      #
+      # @example
+      #
+      #   # bad
+      #   it do
+      #     a?
+      #   end
+      #
+      #   # good
+      #   it do
+      #     expect(a?).to be_truthy
+      #   end
+      class NoExpectationExample < Base
+        extend AutoCorrector
+
+        include RangeHelp
+
+        DEFAULT_EXAMPLE_METHOD_NAMES = ::Set[
+          :example,
+          :fexample,
+          :fit,
+          :focus,
+          :fscenario,
+          :fspecify,
+          :it,
+          :its,
+          :pending,
+          :scenario,
+          :skip,
+          :specify,
+          :xexample,
+          :xit,
+          :xscenario,
+          :xspecify,
+        ]
+
+        DEFAULT_EXPECTATION_METHOD_NAMES = ::Set[
+          :expect_any_instance_of,
+          :expect,
+          :is_expected,
+          :should_not_receive,
+          :should_not,
+          :should_receive,
+          :should,
+        ]
+
+        MSG = 'No expectation found in this example.'
+
+        # @param [RuboCop::AST::Node] node
+        def on_send(node)
+          return unless example_method_names.include?(node.method_name)
+
+          block = argument_block_of(node)
+          return unless block
+          return if including_any_expectation?(block)
+
+          add_offense(block) do |corrector|
+            corrector.remove(removed_range(block))
+          end
+        end
+
+        private
+
+        # @param [RuboCop::AST::Node] node
+        # @return [RuboCop::AST::BlcokNode, nil]
+        def argument_block_of(node)
+          block = node.parent
+          if block.is_a?(::RuboCop::AST::BlockNode) && block.send_node == node
```

Comment:
It looks like it will be replaced with `block_type?` instead of a generic method like `is_a?`.

```suggestion
          if block.block_type? && block.send_node == node
```